src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/TaskExtensions.cs

```diff
@@ -51,6 +51,12 @@ public static bool IsThreadPoolThread(Thread thread)
         }
 #endif
 
+        public static T CompletedResult<T>(this Task<T> task)
+        {
+            Contract.ThrowIfFalse(task.IsCompleted);
+            return task.GetAwaiter().GetResult();
```

Comment:
don't love this extension.  were you intending on using it in more places?

Reply:
Yes, this extension works closely with task usage analysis in vs-threading analyzers (using `Task<T>.Result` is not allowed, but this extension is). I've been meaning to add it for a while.

Reply:
➡️ Improved documentation and added tests